edge-modules/api-proxy-module/templates/nginx_default_config.conf

```diff
@@ -38,25 +38,42 @@ http {
             rewrite ^(.*)$ /storage$1 last;
         }
         #endif_tag ${BLOB_UPLOAD_ROUTE_ADDRESS} 
-        #if_tag !${BLOB_UPLOAD_ROUTE_ADDRESS}
+        #if_tag boolean_expression[!(${BLOB_UPLOAD_ROUTE_ADDRESS})]
         if ($http_x_ms_version)
         {
             rewrite ^(.*)$ /parent$1 last;
         }
-        #endif_tag ${BLOB_UPLOAD_ROUTE_ADDRESS}          
+        #endif_tag boolean_expression[!(${BLOB_UPLOAD_ROUTE_ADDRESS})]          
 
         #if_tag ${DOCKER_REQUEST_ROUTE_ADDRESS}
         if ($uri ~ ^/v2*.)
         {
             rewrite ^(.*)$ /registry$1 last;
         }
         #endif_tag ${DOCKER_REQUEST_ROUTE_ADDRESS}
-        #if_tag !${DOCKER_REQUEST_ROUTE_ADDRESS}
+        #if_tag ${CONNECTED_ACR_ROUTE_ADDRESS}
+        location /v2 {
+            resolver 127.0.0.11;
+            proxy_http_version 1.1;
+            proxy_pass         http://${CONNECTED_ACR_ROUTE_ADDRESS};
+            proxy_set_header   X-Forwarded-Host $http_host;
+            proxy_set_header   X-Forwarded-Proto $scheme;
+        }
+
+        location /acr {
+            resolver 127.0.0.11;
+            proxy_http_version 1.1;
+            proxy_pass         http://${CONNECTED_ACR_ROUTE_ADDRESS};
+            proxy_set_header   X-Forwarded-Host $http_host;
+            proxy_set_header   X-Forwarded-Proto $scheme;
+        }
+        #endif_tag ${CONNECTED_ACR_ROUTE_ADDRESS} 
+        #if_tag boolean_expression[&(!(${DOCKER_REQUEST_ROUTE_ADDRESS}),!(${CONNECTED_ACR_ROUTE_ADDRESS}))]
```

Comment:
Just curious, if the customer defines both DOCKER_REQUEST_ROUTE_ADDRESS and CONNECTED_ACR_ROUTE_ADDRESS for whatever reason, how `/v2` will be routed?

Reply:
This is not supported but this is what would happen:
1. Both tag will be left in the config
2. CONNECTED_ACR_ROUTE_ADDRESS  will route, even if after DOCKER_REQUEST_ROUTE_ADDRESS because it has higher priority. Details are in nginx routing rules.